emukit/core/optimization/local_search_acquisition_optimizer.py

```diff
@@ -0,0 +1,176 @@
+import logging
+from typing import Sequence
+from typing import List
+from typing import Tuple
+from typing import Optional
+
+import numpy as np
+from GPyOpt.optimization.acquisition_optimizer import ContextManager
+
+from .. import ContinuousParameter
+from .. import DiscreteParameter
+from .. import CategoricalParameter
+from .. import OneHotEncoding
+from .. import OrdinalEncoding
+from .. import Parameter
+from .. import ParameterSpace
+from ..acquisition import Acquisition
+from ..optimization.acquisition_optimizer import AcquisitionOptimizerBase
+
+_log = logging.getLogger(__name__)
+
+
+class LocalSearchAcquisitionOptimizer(AcquisitionOptimizerBase):
+    """ Optimizes the acquisition function by greedily following one-exchange neighbours
+        of random initial points.
+
+    One-exchange neighbourhood is defined per parameter type:
```

Comment:
Seems as though other encodings are not supported. Can you mention that in the docstring, as well as a brief comment on how non-categoricals are handled?

Also, is there a paper or another well accepted source which we could reference?

Reply:
I extended the docs describing where it comes from and how parameters are handled. Please check, if it's understandable.